extra_data/tests/test_reader_mockdata.py

```diff
@@ -621,3 +621,28 @@ def test_permission():
         run = RunDirectory(d)
     assert "Permission denied" in str(excinfo.value)
     assert d in str(excinfo.value)
+
+
+def test_filecache(mock_spb_raw_run):
+    fc = get_global_filecache()
+    fc.clean()
+    run = RunDirectory(mock_spb_raw_run)
+    trains_iter = run.trains()
+    tid, data = next(trains_iter)
+    assert tid == 10000
+    device = 'SPB_IRU_CAM/CAM/SIDEMIC:daqOutput'
+    assert device in data
+    assert data[device]['data.image.pixels'].shape == (1024, 768)
+    assert len(fc._cache) == len(run.files)
+    fc.maxfiles = 3
```

Comment:
This will modify the filecache for all tests that run after this one as well. That's not a big problem while everything's working, but it's annoying if we end up with tests that pass or fail depending on what other tests have run before them.

Probably the neatest way to avoid this is to create a new pytest fixture which can reset the global filecache before and after a test which modifies it.

Reply:
I agree.